pkgs/by-name/pi/pixelflasher/package.nix

```
@@ -0,0 +1,101 @@
+{
+  android-tools,
+  cacert,
+  fetchFromGitHub,
+  lib,
+  makeDesktopItem,
+  nix-update-script,
+  python3,
+  stdenv,
+  substituteAll,
+  wrapGAppsHook3,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pixelflasher";
+  version = "7.9.2.4";
+
+  src = fetchFromGitHub {
+    owner = "badabing2005";
+    repo = "PixelFlasher";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-bksTk6Zbj8/4FQ7gUGXfJa+kkJE/bFbJF7A4AY6zIRk=";
+  };
+
+  disabled = python3.pythonOlder "3.11";
+
+  nativeBuildInputs = [ wrapGAppsHook3 ];
+
+  buildInputs = with python3.pkgs; [
+    android-tools
+    attrdict
+    beautifulsoup4
+    bsdiff4
+    chardet
+    cryptography
+    darkdetect
+    httplib2
+    json5
+    lz4
+    markdown
+    platformdirs
+    protobuf
+    psutil
+    pyinstaller
+    pyperclip
+    requests
+    rsa
+    six
+    wxpython
+  ];
+
+  patchPhase = ''
+    # we set the default android-tools path for convenience
+    substituteInPlace config.py --replace-fail \
+      "platform_tools_path = None" "platform_tools_path = '/run/current-system/sw/bin/'"
+  '';
+
+  buildPhase = ''
+    sh build.sh
+  '';
+
+  installPhase = ''
+    install -D dist/PixelFlasher $out/bin/pixelflasher
+    install -D images/icon-dark-256.png $out/share/pixmaps/pixelflasher.png
+    ln -s ${finalAttrs.desktopItem}/share/applications $out/share/
```

```suggestion
    runHook preInstall
    install -D dist/PixelFlasher $out/bin/pixelflasher
    install -D images/icon-dark-256.png $out/share/pixmaps/pixelflasher.png
    runHook postInstall
```
Use `copyDesktopItems` instead